srtcore/core.cpp

```diff
@@ -7348,8 +7421,39 @@ void CUDT::processCtrl(const CPacket &ctrlpkt)
                     // However, this can happen if the packet reordering has caused the earlier sent
                     // LOSSREPORT will be delivered after later sent ACK. Whatever, ACK should be
                     // more important, so simply drop the part that predates ACK.
+                    HLOGC(mglog.Debug, log << CONID() << "LOSSREPORT: adding "
+                            << m_iSndLastAck << "[ACK] - " << losslist_hi << " to loss list");
                     num = m_pSndLossList->insert(m_iSndLastAck, losslist_hi);
                 }
+                else
+                {
```

Comment:
This is a change in the current code that reacts on a situation that wasn't handled before at all. This is a handler for a situation when the lossreport declares a range of packets that as a whole is located behind the current value of `m_iSndLastAck`, that is, refers to packets that have been already dismissed from the sender buffer, or weren't ever in the sender buffer at all. This kind of situation is virtually impossible to happen normally in SRT, however the group code introduce some code based on ISN screwup that adds a risk that this could happen.